test/test_cores/test_fifo/test_ffifo.py

```diff
@@ -80,16 +167,213 @@ def tbstim():
                 yield clock.posedge
                 assert fbus.empty
 
-            # Test overflows        
-            # Test underflows        
-            # Test write / read same time
+            fbus.clear.next = True
+            yield clock.posedge
+            fbus.clear.next = not fbus.clear
+            for ii in range(5):
+                yield clock.posedge
+            raise StopSimulation
+        return tbdut, tbclk, tbstim
+    for kk in range(100):
+        with pytest.raises(ValueError):
+            run_testbench(_bench_fifo_overflow)
+
+# test underflow
+# should give a ValueError
```

Comment:
These pre-function comments can move to the doc-string, one place for global comments/docs for a function.